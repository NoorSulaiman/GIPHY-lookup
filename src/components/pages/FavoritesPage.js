import React, { Component } from 'react';
import { Card } from 'semantic-ui-react';
import GifCardFavorite from '../templates/GifCardFavorite';

class FavoritesPage extends Component {
    state = {}

    componentWillMount() {
        const favorites = JSON.parse(localStorage.getItem("favorites"));
        this.setState({ favorites })
    }

    render() {
        const { favorites } = this.state;
        return (
            <Card.Group stackable itemsPerRow={4}>
                {favorites && favorites.map(gif =>
                    <GifCardFavorite
                        key={gif.id}
                        title={gif.title}
                        imgUrl={gif.imgUrl} />
                )}
            </Card.Group>
        );
    }
}

FavoritesPage.propTypes = {

}



export default FavoritesPage;