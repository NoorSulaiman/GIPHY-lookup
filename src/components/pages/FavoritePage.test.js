import React from 'react';
import { shallow } from 'enzyme';
import { FavoritesPage } from './FavoritesPage';

const clickFn = jest.fn();

describe('FavoritesPage', () => {
    it('should render FavoritesPage correctly with given props', () => {
        const favorites = {};
        const removeFunc = clickFn
        const component = shallow(<FavoritesPage favorites={favorites} removeFunc={removeFunc} />);
        expect(component).toMatchSnapshot();
        expect(component.find('FavoritesGifCard').length).toBe(0)
    });

    it('should render FavoritesGifCard correctly if there is favorites items', () => {
        const favorites = { 'key': { title: 'title', id: 'id', imgUrl: 'value' } };
        const removeFunc = clickFn
        const component = shallow(<FavoritesPage favorites={favorites} removeFunc={removeFunc} />);
        expect(component.find('FavoritesGifCard').length).toBe(1)
    });
})