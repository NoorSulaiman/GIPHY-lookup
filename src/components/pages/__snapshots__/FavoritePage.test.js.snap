// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoritesPage should render FavoritesPage correctly with given props 1`] = `
<CardGroup
  itemsPerRow={4}
  stackable={true}
/>
`;
